src/gui/qgsattributeform.cpp

```diff
@@ -223,14 +223,15 @@ void QgsAttributeForm::setMode( QgsAttributeForm::Mode mode )
   emit modeChanged( mMode );
 }
 
-void QgsAttributeForm::changeAttribute( const QString &field, const QVariant &value )
+void QgsAttributeForm::changeAttribute( const QString &field, const QVariant &value, const QString &toolTip )
 {
   Q_FOREACH ( QgsWidgetWrapper *ww, mWidgets )
   {
     QgsEditorWidgetWrapper *eww = qobject_cast<QgsEditorWidgetWrapper *>( ww );
     if ( eww && eww->field().name() == field )
     {
       eww->setValue( value );
+      eww->widget()->setToolTip( toolTip );
```

Comment:
How about

`eww->setHint( const QString& )` and add a default implementation that does what you do above.

Then in some widgets like "Text Edit" one could do

```
QgsTextEditWidgetWrapper::setHint( const QString& hintText ) override
{
    if ( mBackupText.isNull() )
        mBackupText = hintText;

    if ( hintText.isNull() )
      widget()->setPlaceholderText( mBackupText );
    else
      widget()->setPlaceholderText( hintText );
}
```